### PR TITLE
migration of material files to nullsafety

### PR DESCRIPTION
**Pull request by a14n**
## Description

NNBD migration for part of material

## Related Issues

* dart-lang/sdk#40146

## Tests

I added the following tests:

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/toggleable.dart [hunk 1]
```diff
@@ -410,13 +408,13 @@ abstract class RenderToggleable extends RenderConstrainedBox {
...
     switch (value) {
       case false:
-        onChanged(true);
+        onChanged!(true);
         break;
       case true:
-        onChanged(tristate ? null : false);
+        onChanged!(tristate ? null : false);
         break;
       default: // case null:
```
Comment (Hixie) [inline packages/flutter/lib/src/material/toggleable.dart:416]: can we actually use `case null` now?

**Review by Hixie — APPROVED** [review]
LGTM except for material_state.dart which I'd like @johnsonmh to review.

File: packages/flutter/lib/src/material/toggleable.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter/lib/src/material/toggleable.dart]: Yes, changed.

File: packages/flutter/lib/src/material/material_state.dart [hunk 2]
```diff
@@ -224,15 +222,15 @@ abstract class MaterialStateMouseCursor extends MouseCursor implements MaterialS
   @protected
   @override
   MouseCursorSession createSession(int device) {
-    return resolve(<MaterialState>{}).createSession(device);
+    return resolve(<MaterialState>{})!.createSession(device);
```
Comment (johnsonmh) [inline packages/flutter/lib/src/material/material_state.dart:225]: @dkwingsmt Can you take a look at this? In general `resolve` will return a nullable type, I worry that this `!` isn't safe and should maybe look something more like:

```dart
return resolve(<MaterialState>{})?.createSession(device) ?? _defaultMouseCursor.createSession(device);
```

Comment (HansMuller) [reply packages/flutter/lib/src/material/material_state.dart:225]: Yes. In most (all) cases, if MaterialState's resolve method returns null, that indicates that the widget should defer to the next most general default.

**Review by HansMuller — APPROVED** [review]
LGTM although line 225 in material_state.dart should be reconsidered for a followup change.

**Comment by a14n** [comment]
Could a googler check why _Google testing_ is still _Pending_?

**Comment by renyou** [comment]
We have some infra problem and cannot fetch this PR correctly. Looking into it now.

**Comment by renyou** [comment]
This has something to do with the API we use. I will try to fix this by EOB Monday.

**Pull request merged**